web/controllers/api/docs_controller.ex

```diff
@@ -194,9 +196,38 @@ defmodule HexWeb.API.DocsController do
 
       Ecto.Changeset.change(release, has_docs: false)
       |> HexWeb.Repo.update!
+
+      publish_sitemap()
     end
 
     # TODO: Send mails
     HexWeb.Utils.task(task, fn -> nil end, fn _ -> nil end)
   end
+
+  defp publish_sitemap do
+      # query for has docs packages
+      packages = HexWeb.Package.all_has_docs |> HexWeb.Repo.all
+
+      # build sitemap
+      template =
+        ~s'<?xml version="1.0" encoding="utf-8"?>' <>
+        ~s'<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"' <>
+        ~s'        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"' <>
+        ~s'        xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd">' <>
+        ~s'  <%= for package <- packages do %>' <>
+        ~s'    <url>' <>
+        ~s'      <loc>http://hexdocs.pm/<%= package.name %></loc>' <>
+        ~s'      <lastmod><%=  Ecto.DateTime.to_iso8601(package.updated_at) %></lastmod>' <>
+        ~s'      <changefreq>daily</changefreq>' <>
+        ~s'      <priority>0.8</priority>' <>
+        ~s'    </url>' <>
+        ~s'  <% end %>' <>
+        ~s'</urlset>'
+
+      result = EEx.eval_string template, packages: packages
```

Comment:
We should use `EEx.compile_string` or use phoenix templates because `eval_string` will have bad performance.